packages/core/src/components/way-autosuggest/way-autosuggest.tsx

```diff
@@ -0,0 +1,315 @@
+    this.localSelected.splice(index, 1);
+    this.updateOptions();
+  }
+
+  private handleOnInput = (evt: InputEvent) => {
+    const input = evt.target as HTMLInputElement;
+
+    this.hasFocus = true;
+    this.inputValue = input.value;
+    this.updateOptions();
+  };
+
+  private validationErrors(): string[] {
+    return this.validation ? this.validation(this.localSelected) : [];
+  }
+
+  private calculateMinimumInputWidth(): string {
+    if (!this.inputEl) return '7em';
+
+    const calculation = this.inputEl.value.length * 0.7;
+    if (calculation < 7) {
+      return '7em';
+    } else if (calculation < 15) {
+      return calculation + 'em';
+    } else {
+      return '99%';
+    }
+  }
+
+  private renderTags() {
+    // No tags
+    if (this.localSelected.length === 0) {
+      return null;
+    // If tags not exceeds maxTags
+    } else if (this.localSelected.length < this.localConfig.maxTags + 1) {
+      return (
+        <this.Fragment>
+          {this.localSelected.length > 0 &&
+            this.localSelected.map(tag => (
+              <div title={tag.label} onClick={() => this.removeTag(tag)} class="tag" style={{ backgroundColor: this.config.tagColor }}>
+                <span>{tag.label}</span>
+              </div>
+            ))}
+        </this.Fragment>
+      );
+      // If maxTags is not set or If tags exceed maxTags
+    } else {
+      const text = `${this.localSelected.length} ${this.localConfig.selectedText}`;
+      return (
+        <div class="tag counter" title={text} style={{ backgroundColor: this.config.tagColor }}>
+          <span>{text}</span>
+        </div>
+      );
+    }
+  }
+
+  /**
+   * Used to render multiple HTMLElements without using a wrapper element.
+   * In React JSX we can use '<> <div/><div/><div/> </>' but is not allowed in TS for StencilJS.
+   * 
+   * @param { any } _
+   * @param { HTMLElement[] } children 
+   * 
+   * @returns HTMLElement[]
+   */
+  private Fragment = (_: any, children: HTMLElement[]): HTMLElement[] => [ ...children ];
+
+  componentWillLoad() {
+    if (this.value) {
+      this.localSelected = typeof this.value === 'string' ? JSON.parse(this.value) : this.value;
+    }
+    this.updateOptions();
+    this.renderHiddenInput();
+  }
+
+  render() {
+    return (
+      <Host>
+        <div class={{ 'way-autosuggest': true, 'has-error': this.validationErrors().length > 0 }} onClick={() => this.inputEl && this.inputEl.focus()}>
+          <div class="input-container">
+            {this.renderTags()}
+            <input
+              ref={el => (this.inputEl = el as HTMLInputElement)}
+              class="input"
+              style={{ minWidth: this.calculateMinimumInputWidth() }}
+              type="text"
+              onInput={this.handleOnInput}
+              onClick={() => this.focus()}
+              value={this.inputValue}
+              onFocus={() => this.focus()}
+              placeholder="Start typing"
```

Comment:
**issue**: This will be an issue when used in an application that's not in English. Maybe make a prop of it with a default value of "Start typing"?

Reply:
added placeholder

- [ ] use placeholder prop